packages/store/src/services/store/api/api-client.ts

```diff
@@ -128,4 +138,14 @@
     }
     return undefined
   }
+
+  private extractRequestIdFromErrorMessage(error: unknown): string | undefined {
+    if (error && typeof error === 'object' && 'message' in error) {
+      const message = (error as {message?: string}).message
+      if (message) {
+        return message.match(/Request ID: ([\w-]+)/)?.[1] ?? undefined
+      }
+    }
+    return undefined
```

Comment:
nit, I think it should be OK to parse request ID from the response. However we could have generated it when submitting the request and set request headers. That way we have a guarantee to always have a request ID.